src/FieldRenameHelper.php

```diff
@@ -0,0 +1,134 @@
+<?php
+
+namespace Drupal\localgov_core;
+
+use Drupal\field\Entity\FieldStorageConfig;
+use Drupal\field\Entity\FieldConfig;
+use Drupal\Core\Config\FileStorage;
+use Drupal\Core\Entity\Entity\EntityViewDisplay;
+use Drupal\Core\Entity\Entity\EntityFormDisplay;
+
+class FieldRenameHelper {
+
+  /**
+   * Rename field
+   * @param  string $old_field_name
+   *   Old field machine name.
+   * @param  string $new_field_name
+   *   New field machine name.
+   * @param  string $entity_type
+   *   Entity type name (eg. node, paragraph etc...)
+   */
+  public static function renameField(string $old_field_name, string $new_field_name, string $entity_type) {
+
+    // Rename field.
+    // https://www.drupal.org/docs/drupal-apis/update-api/updating-entities-and-fields-in-drupal-8
+
+    // Make sure new field name is correctly formatted.
+    $new_field_name = substr($new_field_name, 0, 32);
+
+    // Field variables.
+    // @TODO find out how to get the correct table names.
```

Comment:
Perhaps something like this?:
```
$field_storage_key_name = $entity_type . '.field_schema_data.' . $old_field_name;
$key_value_storage = Drupal::keyValue('entity.storage_schema.sql');
$storage_schema = $key_value_storage->get($field_storage_key_name);
$existing_table_names = array_keys($storage_schema);
```

I am trying to find out if we need to update relevant items within the *entity.storage_schema.sql* key's value as part of field renaming.

Reply:
Hi Andy,
I have added some code to copy the fields based on the above.  Hope you don't mind.

I still have some more manual data validation to do.

Reply:
This looks good now so resolving this.